### DISCUSSION TITLE
Additional symbol check every iteration

### DISCUSSION THREAD

**Discussion by fxhuhn** [dodid/minitrade#7] [Q&A]
I'm just learning how to use your great extension and have a question about portfolio customization.

I'm not sure how this works when it comes to multi-asset selection. 

The following idea: I would like to take the strongest values from the NDX100 every month - very similar to your example. How can I show that the symbols in the respective month are also in the index?

```
    def next(self):
        self.alloc.assume_zero()                                #2
        roc = self.roc.df.iloc[-1]                              #3
        (self.alloc.bucket['equity']                            #4
            .append(roc.sort_values(ascending=False), roc > 0)  #5
            .trim(3)                                            #6
            .weight_explicitly(1/3)                             #7
            .apply())                                           #8
        self.rebalance(cash_reserve=0.01)                       #9
```

**Comment by dodid** [comment]
I see what you try to achieve. The historical data used for backtesting are provided at the beginning of a backtest and symbols in the initial data basically define the universe of the run. That means no dynamic data loading or switching from one universe to another at each iteration. One possible way to backtest your strategy is to collect all possible symbols ever appeared in NDX100 and their historical prices, and make that one giant data input.  Supposedly only symbols valid at the iteration should have non-NAN values. You can use that to exclude symbols that are out of life from the index. Alternatively, you can create an indicator in init() to hold the valid index components for each month and use that in next() to screen valid symbols. 

If you talk about trading a strategy with a dynamic universe at every iteration with MT, it's not possible at the moment. I do see a possibility to support that in the incremental mode, say every symbols disappeared from the new universe should be liquidated. But the reality could be more complicated than that.
